test/framework/options.py

```diff
@@ -2968,6 +2968,37 @@ def test_list_software(self):
         self.assertTrue(re.search('^\s+\* GCC v4.6.3: dummy', txt, re.M))
         self.assertFalse(re.search('gzip', txt, re.M))
 
+    def test_parse_optarch(self):
+        """Test correct parsing of optarch option."""
+        
+        options = EasyBuildOptions()
+
+        # Check for EasyBuildErrors
+        error_msg = "The optarch option has an incorrect syntax"
+        options.options.optarch = 'Intel:something;GCC'
+        self.assertErrorRegex(EasyBuildError, error_msg, options.postprocess)
+
+        error_msg = "The optarch option contains duplicated entries for compiler"
+        options.options.optarch = 'Intel:something;GCC:somethingelse;Intel:anothersomething'
+        self.assertErrorRegex(EasyBuildError, error_msg, options.postprocess)
+
+        # Check the parsing itself
+        test_cases = [
+                ('',''),
+                ('xHost','xHost'),
+                ('GENERIC','GENERIC'),
+                ('Intel:xHost', {'Intel': 'xHost'}),
+                ('Intel:GENERIC', {'Intel': 'GENERIC'}),
+                ('Intel:xHost;GCC:march=x86-64 -mtune=generic', {'Intel': 'xHost', 'GCC': 'march=x86-64 -mtune=generic'}),
+                ('Intel:;GCC:march=x86-64 -mtune=generic', {'Intel': '', 'GCC': 'march=x86-64 -mtune=generic'}),
+                ]
```

Comment:
style nitpicking: indent list elements with 4 spaces (no more), single space after `,`, left-align the closing `]`:

```python
test_cases = [
    ('', ''),
    ...
]
```

also, maybe use `gcc_flags = "march=x86-64 -mtune=generic"` to avoid too long lines?

```python
    ('Intel:;GCC:%s' % gcc_flags, {'Intel': '', 'GCC': gcc_flags}),
```